haystack/nodes/answer_generator/openai.py

```diff
@@ -55,12 +53,14 @@ def __init__(
         api_key: str,
         model: str = "text-davinci-003",
         max_tokens: int = 50,
-        top_k: int = 5,
+        top_k: int = 1,
         temperature: float = 0.2,
         presence_penalty: float = 0.1,
         frequency_penalty: float = 0.1,
         examples_context: Optional[str] = None,
         examples: Optional[List[List[str]]] = None,
+        instructions: Optional[str] = None,
+        add_runtime_instructions: bool = False,
```

Comment:
To be honest, I am not really convinced of adding those two params here. It feels like adding "here and there" a piece to instead of adding the needed functionality into the design of PromptTemplate. Finding a suitable design there is not easy but worth the effort, I think. 

I'd propose splitting this PR: 
- one PR that we can merge quickly: changes to the OpenAIError (important!), new stop_words param, clean_documents method
- one PR (or just an issue for now) describing the need / use case you have here for `instructions` and how that potentially look like in a PromptTemplate  
2)

Reply:
I agree, a properly design prompttemplate that allows for runtime instructions is what we need. When do you think we will have such a design + associated testing?

The OpenAIAnswerGenerator class (or its internals) should be substituted by the promptnode + template in the future anyways. That is why I think we should experiment quickly and proceed with the current implementation. If the promptemplate design is just around the corner we dont need to proceed here.

Reply:
There's already a [proposal](https://github.com/deepset-ai/haystack/pull/4172#discussion_r1114361480) out there and it's actively discussed. I would expect to have something implemented in the next weeks. I would rather concentrate our efforts on finding there a nice design that "experimenting on two things in parallel".

Reply:
As the OpenAIErrors are already merged. I'd propose for this PR: Either you reduce it to "new stop_words param" + "clean_documents" OR we close it